test/syncUserFollows.spec.js

```diff
@@ -0,0 +1,133 @@
+const clientErrors = proxies.clientErrors;
+const env = require('./helpers/env');
+const mockAPI = env.USE_MOCK_API;
+const expectOwnProperties = require('./helpers/expectExtensions').expectOwnProperties;
+const baseUrl = config.MYFT_API_URL;
+const extraParams = `?noEvent=${config.MYFT_NO_EVENT}&waitForPurge=${config.MYFT_WAIT_FOR_PURGE_ADD}`;
+const syncUserFollowers = proxies.syncUserFollowers;
+const uuidv4 = require('uuid/v4');
+const syncUserFollowsFix = require('./mocks/fixtures/syncUserFollows');
+const syncConceptFollowsFix = require('./mocks/fixtures/syncConceptFollows');
+const edpFix = require('./mocks/fixtures/emailDigestPreference.json');
+
+const myftConst = config.myftClientConstants;
+const suppressLogs = true; //for local test if you want logs when test are run
+//TODO spec out tests for syncUserFollowers
+//TODO write stubs for:
+
+describe('syncUserFollowers', () => {
+  const fakeGroupId = '00000000-0000-0000-0000-000000000123';
+  const fakeUserId = '00000000-0000-0000-0000-000000000001';
+  const uuid = env.USER_UUID;
+  const groupId = env.LICENCE_UUID;
+
+  const followProps = {
+    byTool:"KAT",
+    byUser:"mock-admin-user"
+  };
+  let getGroupConceptStub;
+  let getGroupSyncedConceptStub;
+  let getConceptsFollowedByUserStub;
+  let addConceptsFollowedByKatUserStub;
+  let getEDPStub;
+  let noGroupConceptsToFollowStub;
+
+  let logMessageStub;
+  const logMessages = [];
+
+  before(done => {
+    if(suppressLogs) {
+      logMessageStub = sinon.stub(logger, 'log').callsFake((...params) => {
+        logMessages.push(params);
+      });
+    }
+
+    done();
+  });
+
+  beforeEach((done) => {
+    //myFT.getConceptsFollowedByGroup
+    getConceptsFollowedByUserStub = sinon.stub(myFT, 'getConceptsFollowedByUser').resolves(syncConceptFollowsFix.userConcepts);
+    addConceptsFollowedByKatUserStub = sinon.stub(myFT, 'addConceptsFollowedByKatUser').resolves(true);
+    getEDPStub = sinon.stub(myFT, 'getEmailDigestPreference').resolves(edpFix);
+    done();
+  });
+
+  afterEach(done => {
+      nock.cleanAll();
+      getConceptsFollowedByUserStub.restore();
+      addConceptsFollowedByKatUserStub.restore();
+      getEDPStub.restore();
+      done();
+  });
+
+  after(done => {
+
+    if(suppressLogs) {
+      logMessageStub.restore();
+    }
+    done();
+  });
+
+  //Happy empty path
+  it('should return status synchronisationIgnored and reason noGroupConceptsToFollow in object for no topics', (done)=> {
+    noGroupConceptsToFollowStub = sinon.stub(myFT, 'getConceptsFollowedByGroup').resolves(syncConceptFollowsFix.noGroupConcepts);
+
+    syncUserFollowers(fakeGroupId, fakeUserId).then(res => {
+      expect(res).to.be.an('object');
+      expect(res).to.have.deep.property('user.status', 'synchronisationIgnored');
+      expect(res).to.have.deep.property('user.reason', 'noGroupConceptsToFollow');
+
+    }).then(() => {
+      //clean up
+      noGroupConceptsToFollowStub.restore();
```

Comment:
Would rather these clean ups be done in an `afterEach` I think. That way, if the Promise throws and doesn't get into this `.then()` we still clean up the stub so that it doesn't stick around and affect another test.

Reply:
Different stubs in different tests so it seemed more appropriate to use stubs where they are needed rather than having a condition in the afterEach() to stop stub.restore() throwing. Removed the extra .then() as thats unecessary. If the promise rejects then the test is going to fail anyway right so does it matter if the following tests blow up as well at that point?